src/transaction/components/TransactionButton.tsx

```diff
@@ -50,41 +52,69 @@ export function TransactionButton({
     transactionId,
   });
 
+  const { errorText, successText } = useMemo(() => {
+    const successText = successOverride?.text
+      ? successOverride?.text
+      : 'View transaction';
+
+    const errorText = errorOverride?.text ? errorOverride?.text : 'Try again';
+
+    return { successText, errorText };
+  }, [errorOverride, successOverride]);
```

Comment:
I also feel like this might be cleaner if the top-level prop is not necessarily an override but the state itself, then we can set the default behavior using a default value:

e.g.
```
success: { text: 'View Transaction', onClick = () => { return window.open(basescan) } };
```

but more of a nit

Reply:
then we can eliminate lines 55-87 (around 30~ LOC)

Reply:
the issue is the default success state can be either `showCallsStatus({ id: transactionId });` or `window.open(...)` depending on wallet type